lib/msf/core/db_manager/import/gpp.rb

```diff
@@ -0,0 +1,36 @@
+require 'rex/parser/group_policy_preferences'
+
+module Msf::DBManager::Import::GPP
+  def import_gpp_xml(args = {}, &block)
+    return unless args && args[:data] && !args[:data].empty?
+
+    gpp = Rex::Parser::GPP.parse(args[:data])
+
+    return unless gpp && gpp.any?
+
+    wspace = find_workspace(args[:workspace])
+
+    return unless wspace && wspace.respond_to?(:id)
+
+    gpp.each do |p|
+      create_credential(
```

Comment:
The end result is still the same, but I think it's interesting that you used the `Auxiliary::Report` methods here instead of the `DbManager` methods directly since you're already in `DbManager`

Reply:
Probably habit.

Reply:
Though I do remember checking this and thought they were coming from `DBManager`:

https://github.com/rapid7/metasploit-framework/blob/682b0863bead6a6972107df1b57cc69452bae5b6/lib/msf/core/auxiliary/report.rb#L32-L38

```
[1] pry(#<Msf::DBManager>)> method(:create_credential)
=> #<Method: Msf::DBManager(Metasploit::Credential::Creation)#create_credential>
[2] pry(#<Msf::DBManager>)>
```

Reply:
Oh yeah, I see it now. I guess I should parse all of the search results instead of just the ones that confirm my suspicions.

Reply:
Same reason I used `report_loot` below instead of `store_loot` from `Msf::Auxiliary::Report`:

```
[2] pry(#<Msf::DBManager>)> method(:report_loot)
=> #<Method: Msf::DBManager(Msf::DBManager::Loot)#report_loot>
[3] pry(#<Msf::DBManager>)>
```

Also, the file to be imported still exists on disk, and its data is stored in the database all the same. It would be helpful to be able to retrieve that data and write it to a file. That's been a long-standing complaint I've had with the `loot` command.

Reply:
Appreciate the review! Thank you!

Reply:
So you actually can retrieve the data and write it to a file with the REST API. If you request the loot from there it returns a `data` field with the file contents base64 encoded.

It's actually stored in the database as well, there's just no easy way to get access to it from `msfconsole` without dropping to `irb`. Definitely something we can add in the future, though.

Reply:
Yeah, most of us use the console. I've been getting the data via `pry` or `irb`, but it's less than ideal. The REST API is nice for programmatic interaction, but I don't see us running `curl` commands every time we want to retrieve loot data.

Reply:
I think it's good we're moving away from the "expert user" experience. I can't reasonably tell someone they should be writing Ruby to retrieve data for a pentest.

Reply:
For posterity...

```
msf5 > irb -e 'puts framework.db.loots.first.data'
<?xml version="1.0" encoding="utf-8"?>
 <Groups   clsid="{3125E937-EB16-4b4c-9934-544FC6D24D26}"
           disabled="1">
   <User   clsid="{DF5F1855-51E5-4d24-8B1A-D9BDE98BA1D1}"
           name="DbAdmin"
           image="2"
           changed="2007-07-06 20:45:20"
           uid="{253F4D90-150A-4EFB-BCC8-6E894A9105F7}">
     <Properties
           action="U"
           newName=""
           fullName="Database Admin"
           description="Local Database Admin"
           cpassword="kHB0+HMUTs/6ySSZ8usxXg=="
           changeLogon="0"
           noChange="0"
           neverExpires="0"
           acctDisabled="1"
           userName="DbAdmin"/>
   </User>
   <Group  clsid="{6D4A79E4-529C-4481-ABD0-F5BD7EA93BA7}"
           name="Database Admins"
           image="2"
           changed="2007-07-06 20:46:21"
           uid="{C5FB3901-508A-4A9E-9171-60D4FC2B404B}">
     <Properties
           action="U"
           newName=""
           description="Local Database Admins"
           userAction="REMOVE"
           deleteAllUsers="1"
           deleteAllGroups="1"
           removeAccounts="0"
           groupName="Database Admins">
       <Members>
         <Member
           name="domain\sampleuser"
           action="ADD"
           sid=""/>
       </Members>
     </Properties>
   </Group>
 </Groups>
msf5 >
```